internal/services/hybridcompute/arc_machine_extension_resource.go

```diff
@@ -0,0 +1,549 @@
+	TypeHandlerVersion     string                              `tfschema:"type_handler_version"`
+}
+
+type MachineExtensionInstanceViewModel struct {
+	Name               string                                    `tfschema:"name"`
+	Status             []MachineExtensionInstanceViewStatusModel `tfschema:"status"`
+	Type               string                                    `tfschema:"type"`
+	TypeHandlerVersion string                                    `tfschema:"type_handler_version"`
+}
+
+type MachineExtensionInstanceViewStatusModel struct {
+	Code          string                             `tfschema:"code"`
+	DisplayStatus string                             `tfschema:"display_status"`
+	Level         machineextensions.StatusLevelTypes `tfschema:"level"`
+	Message       string                             `tfschema:"message"`
+	Time          string                             `tfschema:"time"`
+}
+
+type ArcMachineExtensionResource struct{}
+
+var _ sdk.ResourceWithUpdate = ArcMachineExtensionResource{}
+
+func (r ArcMachineExtensionResource) ResourceType() string {
+	return "azurerm_arc_machine_extension"
+}
+
+func (r ArcMachineExtensionResource) ModelObject() interface{} {
+	return &MachineExtensionModel{}
+}
+
+func (r ArcMachineExtensionResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return machineextensions.ValidateExtensionID
+}
+
+func (r ArcMachineExtensionResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringIsNotEmpty,
+				validation.StringDoesNotContainAny("/"),
+			),
+		},
+
+		"arc_machine_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: machines.ValidateMachineID,
+		},
+
+		"automatic_upgrade_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  true,
+		},
+
+		"force_update_tag": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"location": commonschema.Location(),
+
+		"protected_settings": {
+			Type:             pluginsdk.TypeString,
+			Optional:         true,
+			Sensitive:        true,
+			ValidateFunc:     validation.StringIsJSON,
+			DiffSuppressFunc: pluginsdk.SuppressJsonDiff,
+		},
+
+		"publisher": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"settings": {
+			Type:             pluginsdk.TypeString,
+			Optional:         true,
+			ValidateFunc:     validation.StringIsJSON,
+			DiffSuppressFunc: pluginsdk.SuppressJsonDiff,
+		},
+
+		"type": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
```

Comment:
We don't expose `type` fields, what values can this have? We should potentially split this out into separate resources.